src/OpenTelemetry.Contrib.Extensions.AWSXRay/Resources/AWSEKSResourceDetector.cs

```diff
@@ -0,0 +1,148 @@
+        public IEnumerable<KeyValuePair<string, object>> Detect()
+        {
+            List<KeyValuePair<string, object>> resourceAttributes = null;
+
+            try
+            {
+                var clusterName = this.GetEKSClusterName();
+                var containerId = this.GetEKSContainerId();
+
+                if (clusterName == null && containerId == null)
+                {
+                    return resourceAttributes;
+                }
+
+                resourceAttributes = new List<KeyValuePair<string, object>>()
+                {
+                    new KeyValuePair<string, object>(AWSSemanticConventions.AttributeCloudProvider, "aws"),
+                    new KeyValuePair<string, object>(AWSSemanticConventions.AttributeCloudPlatform, "aws_eks"),
+                    new KeyValuePair<string, object>(AWSSemanticConventions.AttributeK8SClusterName, clusterName),
+                    new KeyValuePair<string, object>(AWSSemanticConventions.AttributeContainerID, containerId),
+                };
+            }
+            catch (Exception ex)
+            {
+                AWSXRayEventSource.Log.ResourceAttributesExtractException(nameof(AWSEKSResourceDetector), ex);
+            }
+
+            return resourceAttributes;
+        }
+
+        private string GetEKSClusterName()
+        {
+            var credentials = this.GetEKSCredentials();
+
+            if (!this.IsEKSProcess(credentials))
+            {
+                return null;
+            }
+
+            var clusterInfo = this.GetEKSClusterInfo(credentials);
+            var clusterInfoObject = this.DeserializeResponse(clusterInfo);
+
+            return clusterInfoObject.Data?.ClusterName;
+        }
+
+        private bool IsEKSProcess(string credentials)
+        {
+            var httpClientHandler = this.CreateHttpClientHandler();
+            var awsAuth = ResourceDetectorUtils.SendOutRequest(AWSAuthUrl, "GET", new KeyValuePair<string, string>("Authorization", credentials), httpClientHandler).Result;
+            return string.IsNullOrEmpty(awsAuth);
+        }
+
+        private string GetEKSClusterInfo(string credentials)
+        {
+            var httpClientHandler = this.CreateHttpClientHandler();
+            return ResourceDetectorUtils.SendOutRequest(AWSClusterInfoUrl, "GET", new KeyValuePair<string, string>("Authorization", credentials), httpClientHandler).Result;
+        }
+
+        private HttpClientHandler CreateHttpClientHandler()
+        {
+            var httpClientHandler = new HttpClientHandler();
+            httpClientHandler.ClientCertificates.Add(new X509Certificate2(AWSEKSCertificatePath));
+            return httpClientHandler;
+        }
+
+        private AWSEKSClusterInformationModel DeserializeResponse(string response)
+        {
+            return ResourceDetectorUtils.DeserializeFromString<AWSEKSClusterInformationModel>(response);
+        }
+
+        private string GetEKSCredentials()
```

Comment:
You can test these with fake or mocked files

Reply:
Added